src/app/pages/Swap/modals/SwapSettings/index.tsx

```diff
@@ -41,16 +42,16 @@ const SwapSettingsModal = memo<SelectTokenModalProps>(({ onSubmit, opened, onReq
 
   const handleOptionChange = (option: (typeof options)[number]) => {
     setSelected(option);
-    if (option === 'Custom') {
+    if (option === customLabel) {
       setValue('slippageTolerance', undefined, { shouldValidate: formSubmitted });
-    } else if (option === 0.5 || option === 1) {
-      setValue('slippageTolerance', option, { shouldValidate: formSubmitted });
+    } else if (options.includes(option)) {
+      setValue('slippageTolerance', option as number, { shouldValidate: formSubmitted });
```

Comment:
I suppose checking of the type of `option` instead of comparing it to value is a good option for removing typecasts like this.